appium/tests/specs/mock/setup/CheckRevokedKeyUpdate.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { ekmKeySamples } from 'api-mocks/apis/ekm/ekm-endpoints';
+import { MockApi } from 'api-mocks/mock';
+import {
+  SplashScreen
+} from '../../../screenobjects/all-screens';
+import MailFolderScreen from "../../../screenobjects/mail-folder.screen";
+import NewMessageScreen from "../../../screenobjects/new-message.screen";
+import SetupKeyScreen from "../../../screenobjects/setup-key.screen";
+
+describe('SETUP: ', () => {
+
+  it('check revoked key update logic', async () => {
+
+    const mockApi = new MockApi();
+    const contactEmail = 'available.on@attester.test';
+    const contactName = 'Test1';
+
+    mockApi.fesConfig = {
+      clientConfiguration: {
+        flags: ["NO_PRV_CREATE", "NO_PRV_BACKUP", "NO_ATTESTER_SUBMIT", "PRV_AUTOIMPORT_OR_AUTOGEN", "FORBID_STORING_PASS_PHRASE"],
+        key_manager_url: "https://ekm.flowcrypt.com"
+      }
+    };
+    mockApi.attesterConfig = {
+      servedPubkeys: {
+        [contactEmail]: ekmKeySamples.key0Revoked.pub!
+      }
+    };
+
+    await mockApi.withMockedApis(async () => {
+      await SplashScreen.login();
+      await SetupKeyScreen.setPassPhrase();
+      await MailFolderScreen.checkInboxScreen();
+      await MailFolderScreen.clickCreateEmail();
+
+      await NewMessageScreen.setAddRecipient(contactEmail);
+      await NewMessageScreen.checkAddedRecipientColor(contactName, 0, 'red');
+
+      mockApi.attesterConfig = {
+        servedPubkeys: {
+          [contactEmail]: ekmKeySamples.key0Updated.pub!
+        }
+      };
+
+      await NewMessageScreen.deleteAddedRecipient(0);
+
+      // Should display name in red background(which means revoked key) even though we receive new updated key
+      // Because we already have revoked key on local
+      await NewMessageScreen.setAddRecipient(contactEmail);
+      await NewMessageScreen.checkAddedRecipientColor(contactName, 0, 'red');
```

Comment:
I think the implementation broke general key update functionality. You can check it by adding the following steps at the end:
 - delete added recipient
 - set Attester to some other unrelated public key that is valid
 - add recipient again, and wait for it to update (maybe add two seconds to give it a chance to update)

You should see that the recipient should turn green. But as implemented, I think it will wrongly stay red.

Reply:
I added test and it works fine.